localization/ekf_localizer/src/extended_kalman_filter_module.cpp

```diff
@@ -0,0 +1,319 @@
+// Copyright 2018-2019 Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ekf_localizer/extended_kalman_filter_module.hpp"
+
+#include "ekf_localizer/covariance.hpp"
+#include "ekf_localizer/mahalanobis.hpp"
+#include "ekf_localizer/matrix_types.hpp"
+#include "ekf_localizer/measurement.hpp"
+#include "ekf_localizer/numeric.hpp"
+#include "ekf_localizer/state_transition.hpp"
+#include "ekf_localizer/warning_message.hpp"
+
+#include <tier4_autoware_utils/geometry/geometry.hpp>
+#include <tier4_autoware_utils/ros/msg_covariance.hpp>
+
+#include <fmt/core.h>
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2/utils.h>
+
+// clang-format off
+#define DEBUG_PRINT_MAT(X) {\
+  if (params_.show_debug_info) {std::cout << #X << ": " << X << std::endl;}\
+}
+// clang-format on
+
+ExtendedKalmanFilterModule::ExtendedKalmanFilterModule(
+  std::shared_ptr<Warning> warning, const HyperParameters params)
+: warning_(std::move(warning)),
+  dim_x_(6),  // x, y, yaw, yaw_bias, vx, wz
+  params_(params)
+{
+  Eigen::MatrixXd X = Eigen::MatrixXd::Zero(dim_x_, 1);
+  Eigen::MatrixXd P = Eigen::MatrixXd::Identity(dim_x_, dim_x_) * 1.0E15;  // for x & y
+  P(IDX::YAW, IDX::YAW) = 50.0;                                            // for yaw
+  if (params_.enable_yaw_bias_estimation) {
+    P(IDX::YAWB, IDX::YAWB) = 50.0;  // for yaw bias
+  }
+  P(IDX::VX, IDX::VX) = 1000.0;  // for vx
+  P(IDX::WZ, IDX::WZ) = 50.0;    // for wz
+
+  kalman_filter_.init(X, P, params_.extend_state_step);
+}
+
+void ExtendedKalmanFilterModule::initialize(
+  PoseWithCovariance & initial_pose, geometry_msgs::msg::TransformStamped & transform)
```

Comment:
You can make it const argument. (Of course, hpp needs to be changed as well.)
```suggestion
void ExtendedKalmanFilterModule::initialize(
  const PoseWithCovariance & initial_pose, const geometry_msgs::msg::TransformStamped & transform)
```

Reply:
Fixed here: https://github.com/autowarefoundation/autoware.universe/pull/5424/commits/048faf9e4767469a200a71d37957dcd39469191b